pkg/version/remote.go

```diff
@@ -142,9 +153,64 @@ func WithReference(current semver.Version) CheckOption {
 	})
 }
 
+// cachedCheck returns previously saved result from cache
+func cachedCheck() (*cachedUpdate, error) {
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return nil, err
+	}
+	cacheFile := filepath.Join(cacheDir, "tts", "version-check")
```

Comment:
For the sake of DRY, maybe pull this into a little func? Then you can also fall back to `os.TempDir` on error.

You can ever consider making it a var:

```go
var cacheFile = func() string {
	cacheDir, err := os.UserCacheDir()
	if err != nil {
		cacheDir = os.TempDir()
	}
	return filepath.Join(cacheDir, "tts", "version-check")
}()
```